internal/server/service.go

```diff
@@ -128,10 +128,14 @@ func (s *Server) CreateEntry(ctx context.Context, req *pb.CreateEntryRequest) (*
 	if errors.Is(err, ttessera.ErrPushback) {
 		return nil, status.Errorf(codes.Unavailable, "reached max pushback; retry")
 	}
-	if errors.As(err, &tessera.DuplicateError{}) {
+	if errors.Is(err, context.Canceled) {
+		// Returns a 499 Client Closed Request
+		return nil, status.Error(codes.Canceled, err.Error())
+	}
+	if errors.Is(err, tessera.DuplicateError{}) {
```

Comment:
I think you can use the type directly, instead of the empty struct.

Reply:
Using just `tessera.DuplicateError` doesn't compile, though there's an issue with this change, the e2e tests are failing. Looking into it...

Reply:
I think `errors.Is` is for comparing an `error`, rather than `errors.As` which is for custom types that implement the `error` interface.